WalletWasabi.Tests/UnitTests/JsonConverters/FeeRateJsonConverterTests.cs

```diff
@@ -0,0 +1,167 @@
+using NBitcoin;
+using Newtonsoft.Json;
+using Newtonsoft.Json.Linq;
+using WalletWasabi.JsonConverters.Bitcoin;
+using Xunit;
+using JsonConvertNew = System.Text.Json.JsonSerializer;
+using JsonConvertOld = Newtonsoft.Json.JsonConvert;
+
+namespace WalletWasabi.Tests.UnitTests.JsonConverters;
+
+public class FeeRateJsonConverterTests
+{
+	/// <summary>
+	/// Tests that JSON converter based on <c>System.Text.Json</c> and the one based on <c>Newtonsoft.Json</c> *serialize* objects equally.
+	/// </summary>
+	[Fact]
+	public void SerializationParity()
+	{
+		TestData testObject = new();
+
+		string json = AssertSerializedEqually(testObject);
+		Assert.Equal("""{"FeePaidSize":500000,"FeePerKByte":1000,"SatoshiPerByte":1000000,"Zero":0,"NoneFeeRate":null,"NotAnnotated":null}""", json);
+	}
+
+	/// <summary>
+	/// Tests that JSON converter based on <c>System.Text.Json</c> and the one based on <c>Newtonsoft.Json</c> *deserialize* objects equally.
+	/// </summary>
+	[Fact]
+	public void DeserializationParity()
+	{
+		// Success cases.
+		{
+			string token = "0";
+			AssertBothDeserialize(token);
+
+			token = "1";
+			AssertBothDeserialize(token);
+
+			token = "209999999"; // Maximum number of bitcoins ever to exist + 1 satoshi.
+			AssertBothDeserialize(token);
+
+			token = "9223372036854775807"; // Biggest long number
+			AssertBothDeserialize(token);
+		}
+
+		// Format exception errors.
+		{
+			// The old converter throws InvalidCastException, the new converter throws JsonException.
+			string invalidToken = "1e6";
+			AssertDeserializeDifferentExceptions<InvalidCastException, System.Text.Json.JsonException>(invalidToken);
+
+			// The old converter throws JsonReaderException, the new converter throws JsonException.
+			invalidToken = "1,0";
+			AssertDeserializeDifferentExceptions<JsonReaderException, System.Text.Json.JsonException>(invalidToken);
+
+			// The old converter throws JsonReaderException, the new converter throws JsonException.
+			invalidToken = "1,000.00";
+			AssertDeserializeDifferentExceptions<JsonReaderException, System.Text.Json.JsonException>(invalidToken);
+		}
+
+		// Unique case.
+		{
+			string invalidToken = "1."; // No digit after decimal point.
+			AssertDeserializeDifferentExceptions<InvalidCastException, System.Text.Json.JsonException>(invalidToken);
+		}
+
+		// Tests that neither JSON converters can deserialize to NULL if a JSON number-string is found instead of a JSON integer.
+		{
+			string invalidToken = "100";
+			AssertDeserializeDifferentExceptions<InvalidCastException, System.Text.Json.JsonException>(ConvertToJsonString(invalidToken));
```

Comment:
This raised a question in me about why is this a failing case.

```suggestion
			string invalidToken = ConvertToJsonString("100");
			AssertDeserializeDifferentExceptions<InvalidCastException, System.Text.Json.JsonException>(invalidToken);
```

Perhaps this suggestion can clarify why this is a failing case for future reviewers and devs.